android-reactive-location/src/main/java/pl/charmas/android/reactivelocation/observables/BaseObservable.java

```diff
@@ -30,7 +30,20 @@ protected BaseObservable(Context ctx, Api<? extends Api.ApiOptions.NotRequiredOp
     @Override
     public void call(Subscriber<? super T> subscriber) {
 
-        final GoogleApiClient apiClient = createApiClient(subscriber);
+        final ApiClientConnectionCallbacks apiClientConnectionCallbacks = new ApiClientConnectionCallbacks(subscriber);
+
+        GoogleApiClient.Builder apiClientBuilder = new GoogleApiClient.Builder(ctx);
+
+        for (Api<? extends Api.ApiOptions.NotRequiredOptions> service : services) {
+            apiClientBuilder.addApi(service);
+        }
+
+        apiClientBuilder.addConnectionCallbacks(apiClientConnectionCallbacks);
+        apiClientBuilder.addOnConnectionFailedListener(apiClientConnectionCallbacks);
+
+        final GoogleApiClient apiClient = apiClientBuilder.build();
+        apiClientConnectionCallbacks.setClient(apiClient);
+
```

Comment:
The setup isn't actually different from the deleted `createApiClient` right? If yes, I think we should leave it as it was.